src/jit/flowgraph.cpp

```diff
@@ -13652,11 +13720,17 @@ bool Compiler::fgOptimizeSwitchBranches(BasicBlock* block)
         switchTree->gtOp.gtOp1  = condNode;
         switchTree->gtOp.gtOp1->gtFlags |= (GTF_RELOP_JMP_USED | GTF_DONT_CSE);
 
-        // Re-link the nodes for this statement.
-        // We know that this is safe for the Lowered form, because we will have eliminated any embedded trees
-        // when we cloned the switch condition (it is also asserted above).
+        if (block->IsLIR())
+        {
+            blockRange.InsertAfter(zeroConstNode, switchVal);
+            blockRange.InsertAfter(condNode, zeroConstNode);
+        }
```

Comment:
No: the nodes to be inserted are consuming a tree that is already in the block and therefore cannot be resequenced.